tools/ci/run_tc.py

```diff
@@ -0,0 +1,261 @@
+or the branch that the commits are on (if it's a push), and GITHUB_PULL_REQUEST
+which is the string "false" if the event triggering this job wasn't a pull request
+or the pull request number if it was. The semantics of these variables are chosen
+to match the corresponding TRAVIS_* variables.
+
+Note: for local testing in the Docker image the script ought to still work, but
+full functionality requires that the TASK_EVENT environment variable is set to
+the serialization of a GitHub event payload.
+"""
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import sys
+try:
+    from urllib2 import urlopen
+except ImportError:
+    # Python 3 case
+    from urllib.request import urlopen
+
+
+root = os.path.abspath(
+    os.path.join(os.path.dirname(__file__),
+                 os.pardir,
+                 os.pardir))
+
+
+def run(cmd, return_stdout=False, **kwargs):
+    print(" ".join(cmd))
+    if return_stdout:
+        f = subprocess.check_output
+    else:
+        f = subprocess.check_call
+    return f(cmd, **kwargs)
+
+
+def start(cmd):
+    print(" ".join(cmd))
+    subprocess.Popen(cmd)
+
+
+def get_parser():
+    p = argparse.ArgumentParser()
+    p.add_argument("--oom-killer",
+                   action="store_true",
+                   default=False,
+                   help="Run userspace OOM killer")
+    p.add_argument("--hosts",
+                   dest="hosts_file",
+                   action="store_true",
+                   default=True,
+                   help="Setup wpt entries in hosts file")
+    p.add_argument("--no-hosts",
+                   dest="hosts_file",
+                   action="store_false",
+                   help="Don't setup wpt entries in hosts file")
+    p.add_argument("--browser",
+                   action="append",
+                   default=[],
+                   help="Browsers that will be used in the job")
+    p.add_argument("--channel",
+                   default=None,
+                   choices=["experimental", "dev", "nightly", "beta", "stable"],
+                   help="Chrome browser channel")
+    p.add_argument("--xvfb",
+                   action="store_true",
+                   help="Start xvfb")
+    p.add_argument("--checkout",
+                   help="Revision to checkout before starting job")
+    p.add_argument("job",
+                   help="Name of the job associated with the current event")
+    p.add_argument("script",
+                   help="Script to run for the job")
+    p.add_argument("script_args",
+                   nargs=argparse.REMAINDER,
+                   help="Additional arguments to pass to the script")
+    return p
+
+
+def start_userspace_oom_killer():
+    # Start userspace OOM killer: https://github.com/rfjakob/earlyoom
+    # It will report memory usage every minute and prefer to kill browsers.
+    start(["sudo", "earlyoom", "-p", "-r", "60" "--prefer=(chrome|firefox)", "--avoid=python"])
```

Comment:
There's a missing comma here, sent https://github.com/web-platform-tests/wpt/pull/18630 to fix that.